src/vm/tieredcompilation.cpp

```diff
@@ -201,6 +201,10 @@ void TieredCompilationManager::AsyncPromoteMethodToTier1(MethodDesc* pMethodDesc
         }
```

Comment:
Would it be helpful to add logging to any of the failure cases above, where we fail due to it already being optimized, OOM, etc?

Reply:
I was mostly just adding logging specific to what I needed to debug, but I went ahead and added those too. I also noticed one of the existing log statements wasn't quite in the right spot so I moved it while I was in there.